cognite/client/config.py

```diff
@@ -46,6 +59,43 @@ def __init__(self) -> None:
         self.max_workers: int = 5
         self.silence_feature_preview_warnings: bool = False
 
+    def apply_settings(self, settings: dict[str, Any] | str) -> None:
+        """Apply settings to the global configuration object from a YAML/JSON string or dict.
+
+        Warning:
+            This must be done before instantiating a CogniteClient for the configuration to take effect.
+
+        Args:
+            settings (dict[str, Any] | str): A dictionary or YAML/JSON string containing configuration values defined in the GlobalConfig class.
+
+        Examples:
+
+            Create a global config object from a dictionary input:
+
+                >>> from cognite.client import global_config
+                >>> settings = {
+                ...     "max_retries": 5,
+                ...     "disable_ssl": True,
+                ... }
+                >>> global_config.apply_settings(settings)
+        """
+
+        loaded = load_resource_to_dict(settings)
+
+        if not loaded.keys() <= self.__dict__.keys():
+            raise ValueError(
+                f"One or more invalid keys provided for global_config, no settings applied: {set(loaded) - set(self.__dict__)}"
+            )
+
+        for key, value in loaded.items():
+            if key == "default_client_config":
+                if isinstance(value, ClientConfig):
+                    self.default_client_config = value
+                else:
+                    self.default_client_config = ClientConfig.load(value)
```

Comment:
Will fix 👍🏻 but that raises another question because current implementation could then partially apply settings if you provide an invalid value to any of the other types like passing a dict to max_retries or something. So should probably validate all the fields then before applying them

Reply:
No, all the others are just set as attributes and this won't raise